cmd/mailchain/internal/http/handlers/messages.go

```diff
@@ -98,9 +98,13 @@ func GetMessages(inbox stores.State, receivers map[string]mailbox.Receiver, ks k
 					MessageID:   message.ID.HexString(),
 					ContentType: message.Headers.ContentType,
 				},
-				Read:    readStatus,
-				Subject: message.Headers.Subject,
-				Status:  "ok",
+				Read:                    readStatus,
+				Subject:                 message.Headers.Subject,
+				Status:                  "ok",
+				BlockNumber:             transactionData.BlockNumber,
+				BlockNumberEncoding:     transactionData.EncodingType,
+				TransactionHash:         transactionData.Hash,
+				TransactionHashEncoding: transactionData.EncodingType,
```

Comment:
Because encoding type is not returned by receiver we need to set it here, I would suggest return a fixed encoding for now e.g. `encoding.TypeHex0XPrefix`. 
At a later date we can look at returning by protocol or depending on what the client requests.

One other issue that is present though can be see by looking at https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_getblockbyhash `block-number` is a `big.Int` and `transaction-hash` is a `byte` this needs to be exposed some how.